src/Account/StudentTopicPreferences.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, FormControlLabel, Grid, Switch } from '@material-ui/core';
+import React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import _ from 'lodash';
+import { nameof } from '../Utilities/TypescriptUtils';
+import localPreferences from '../Utilities/LocalPreferences';
+import useAlertState from '../Hooks/useAlertState';
+import { Alert } from 'react-bootstrap';
+
+const { topicPreferences } = localPreferences;
+
+export interface StudentTopicPreferencesInputs {
+    useSeconds: boolean;
+}
+
+interface StudentTopicPreferencesProps {
+
+}
+
+
+export const StudentTopicPreferences: React.FC<StudentTopicPreferencesProps> = () => {
+    const [alert, setAlert] = useAlertState();
+    const preferences = useForm<StudentTopicPreferencesInputs>({
+        mode: 'onSubmit',
+        shouldFocusError: true,
+        defaultValues: {
+            useSeconds: topicPreferences.useSeconds
+        }
+    });
+    const { control, getValues, handleSubmit, watch } = preferences;
+    const useSeconds = watch('useSeconds');
+
+    const onSubmit = (values: StudentTopicPreferencesInputs) => {
+        topicPreferences.useSeconds = values.useSeconds;
+        setAlert({
+            variant: 'success',
+            message: 'Saved Successful'
+        });
+    };
+
+    const clearAlert = () => {
+        setAlert({
+            variant: 'info',
+            message: ''
+        });
+    };
+
+    return (<Grid container item spacing={3} xs={6} justify='center'>
+        <form onChange={() => clearAlert()} onSubmit={handleSubmit(onSubmit)}>
+            <Alert variant={alert.variant} show={alert.message.length > 0}>{alert.message}</Alert>
```

Comment:
We could probably simplify all our alerts to only show if there's a message set. Nice.

Reply:
I think I grabbed it from your code 🤣